views/fe-utils/preferencesUtil.js

```diff
@@ -0,0 +1,21 @@
+export const babyProofPreferences = (preferences) => {
+  let preference = {}
+  preference['star'] = 'star' in preferences ? preferences['star'] : {}
+  preference['sort'] = 'sort' in preferences ? preferences['sort'] : 0
+  preference['config'] = 'config' in preferences ? preferences['config'] : ''
+  preference['complete'] =
+    'complete' in preferences ? preferences['complete'] : {}
+  return preference
+}
+
+export const preparePreferences = (configurations, preferences) => {
+  console.log(configurations, preferences)
+  let preference = {}
+  preference['config'] = configurations._id
+  preference['complete'] =
+    'complete' in preferences ? preferences['complete'] : {}
+  preference['star'] = 'star' in preferences ? preferences['star'] : {}
+  preference['sort'] = 'sort' in preferences ? preferences['sort'] : 0
+  preference = babyProofPreferences(preference)
+  return preference
+}
```

Comment:
one more question here, when we build `preferences` we set a key of `config` but it's value is just an id, should that key be named `id` or `config_id`? Alternatively, If we plan to have more configurations as part of `config` should it be an object that contains an key/value of `id: _id`

Reply:
It could be renamed to like configId because it's not obvious that it's supposed to be a config id. Something to refactor in a future iteration.

https://github.com/AMP-SCZ/dpdash/pull/129/commits/ac406194c712160522e6718ec0e8a81a59939cd0

Reply:
that change looks good for the body of `babyProofPreferences`, though it looks like we still have a lot of logic in `preparePreferences` that could be removed since it calls `babyProofPreferences` to provide those default values

Reply:
That's a good point. What do you suggest?

Reply:
> There's lot's of weirdness that happens... like using an async await function outside of the component to post to preferences results in an empty json body.

I'm not sure I follow.
Just pointing out that with the exception of `config`, everything in `preparePreferences` is already handled by `babyProofPreferences` which `preparePreferences` calls, so I think we're providing those defaults twice and we could simplify `preparePreferences` to the following and have the same functionality. I don't have the same context as you though 😄 so if I'm missing something and there's a reason to keep it the way it is please do 👍 
```js
export const preparePreferences = (configurationId = '', preferences = {}) => 
  ({ ...babyProofPreferences(preferences), config: configurationId })
```

Reply:
Thanks for clearing that up. https://github.com/AMP-SCZ/dpdash/pull/129/commits/beb63138114b0ed686df00274fbdb8372ed4e84d